### PR TITLE
feat: code optimizations and execute multiple trades

### PR DESCRIPTION
**Pull request by lbertenasco**

### CODE REVIEW

File: contracts/Swapper.sol [hunk 1]
```diff
@@ -7,7 +7,9 @@ import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
 import '@lbertenasco/contract-utils/contracts/utils/Governable.sol';
 import '@lbertenasco/contract-utils/contracts/utils/CollectableDust.sol';
 
-interface ISwapper {
+import './utils/ITrade.sol';
+
+interface ISwapper is ITrade {
```
[the remark this answers is not in the document]
Comment (lbertenasco) [reply contracts/Swapper.sol:12]: True, i'll change it

File: contracts/TradeFactory/TradeFactoryPositionsHandler.sol [hunk 2]
```diff
@@ -119,7 +109,7 @@ abstract contract TradeFactoryPositionsHandler is ITradeFactoryPositionsHandler,
     require(_tokenIn != address(0) && _tokenOut != address(0), 'TradeFactory: zero address');
     require(_amountIn > 0, 'TradeFactory: zero amount');
     require(_maxSlippage > 0, 'TradeFactory: zero slippage');
-    require(block.timestamp < _deadline, 'TradeFactory: deadline too soon');
+    require(_deadline > block.timestamp, 'TradeFactory: deadline too soon');
```
[the remark this answers is not in the document]
Comment (lbertenasco) [reply contracts/TradeFactory/TradeFactoryPositionsHandler.sol:112]: OCD

**Comment by b0dhidharma** [comment]
I have canibalized everything from this PR :joy:  ... so its kinda merged hehe. (Stolen the script for executing trades also)

**Comment by b0dhidharma** [comment]
I have canibalized everything from this PR :joy:  ... so its kinda merged hehe :. (Stolen the script for executing trades also)

**Pull request closed without merging**